File: examples/Calc/CalcFeatures/CalculatorSteps.cs
using System;
using Calc;
using Cuke4Nuke.Framework;
using NUnit.Framework;

namespace CalcFeatures
{
    class CalculatorSteps
    {
        private Calculator _calculator;
        private double _result;

        [Before]
        public void CreateCalculator()
        {
            _calculator = new Calculator();
        }

        [Given(@"^I have entered (\d+) into the calculator$")]
        public void EnterNumber(double n)
        {
            _calculator.Push(n);
        }

        [When(@"^I press divide$")]
        public void Divide()
        {
            _result = _calculator.Divide();
        }

        [When(@"^I press add$")]
        public void Add()
        {
            _result = _calculator.Add();
        }

        [Then(@"^the result should be (.*) on the screen$")]
        public void CheckResult(double expected)
        {
            Assert.That(_result, Is.EqualTo(expected));
        }
    }
}

File: examples/Calc/CalcFeatures/features/step_definitons/calculator_steps.rb
if Cucumber::IRONRUBY # We only want this to run on IronRuby

require 'spec/expectations'
$:.unshift(File.dirname(__FILE__) + '/../../bin/Debug') # So we find the .dll
require 'Calc.dll'

Before do
  @calc = Calc::Calculator.new
end

Given "I have entered $n into the calculator" do |n|
  @calc.Push(n.to_f)
end

When /I press divide/ do
  @result = @calc.Divide
end

When /I press add/ do
  @result = @calc.Add
end

Then /the result should be (.*) on the screen/ do |result|
  @result.should == result.to_f
end

end